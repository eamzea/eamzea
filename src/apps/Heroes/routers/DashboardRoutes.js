import React from 'react';
import NavbarHeroes from '../components/ui/Navbar';
import { Switch, Route, Redirect, useRouteMatch } from 'react-router-dom';
import MarvelScreen from '../components/marvel/MarvelScreen';
import HeroScreen from '../components/heroes/HeroScreen';
import DcScreen from '../components/dc/DcScreen';
import { Container } from 'react-bootstrap';
import SearchScreen from '../components/search/SearchScreen';
import FloatLink from '../../Portfolio/components/ui/FloatLink';

const DashboardRoutes = () => {
  const { path } = useRouteMatch();

  return (
    <>
      <NavbarHeroes />
      <Container fluid>
        <Switch>
          <Route exact path={`${path}/marvel`} component={MarvelScreen} />
          <Route exact path={`${path}/hero/:heroId`} component={HeroScreen} />
          <Route exact path={`${path}/dc`} component={DcScreen} />
          <Route exact path={`${path}/search`} component={SearchScreen} />
          <Redirect to={`${path}/marvel`} />
        </Switch>
        <FloatLink />
      </Container>
    </>
  );
};

export default DashboardRoutes;
