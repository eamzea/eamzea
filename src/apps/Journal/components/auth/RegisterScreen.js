import React from 'react';
import { Link } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { setError, removeError } from '../../actions/ui';
import { startRegisterEmailPasswordName } from '../../actions/auth';

const RegisterScreen = () => {
  const dispatch = useDispatch();

  const { msgError } = useSelector(state => state.ui);

  const [formValues, handleInputChange] = useForm({
    name: 'Edgar',
    email: 'eamzea@gmail.com',
    password: 'cesarZEA210394',
    password2: 'cesarZEA210394',
  });

  const { name, email, password, password2 } = formValues;

  const handleRegister = e => {
    e.preventDefault();

    if (isFormValid()) {
      dispatch(startRegisterEmailPasswordName(email, password, name));
    }
  };

  const isFormValid = () => {
    if (name.trim().length === 0) {
      dispatch(setError('Name is Required'));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(setError('Email is not valid'));
      return false;
    } else if (password.length <= 5) {
      dispatch(setError('Password should be at least 6 characters'));
      return false;
    } else if (password !== password2) {
      dispatch(setError(`Passwords don't match`));
      return false;
    }

    dispatch(removeError());

    return true;
  };

  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form
        onSubmit={handleRegister}
        className="animate__animated animate__fadeIn animate__faster"
      >
        {msgError && <div className="auth__alert-error">{msgError}</div>}

        <input
          className="auth__input"
          type="text"
          placeholder="Name"
          name="name"
          autoComplete="off"
          value={name}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="text"
          placeholder="Email"
          name="email"
          autoComplete="off"
          value={email}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Password"
          name="password"
          value={password}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Confirm password"
          name="password2"
          value={password2}
          onChange={handleInputChange}
        />
        <button type="submit" className="btn btn-primary btn-block mb-5">
          Register
        </button>

        <Link to="/journal/auth/login" className="link">
          Already registered?
        </Link>
      </form>
    </>
  );
};

export default RegisterScreen;
