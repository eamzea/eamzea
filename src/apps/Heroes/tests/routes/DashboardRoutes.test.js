import "@testing-library/jest-dom";
import React from "react";
import { shallow, mount } from "enzyme";
import DashboardRoutes from "../../routers/DashboardRoutes";
import { AuthContext } from "../../auth/AuthContext";
import { MemoryRouter } from "react-router-dom";

describe("Pruebas en <DashboardRoutes/>", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Edgar",
    },
  };

  test("Debe mostrarse correctamente", () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
