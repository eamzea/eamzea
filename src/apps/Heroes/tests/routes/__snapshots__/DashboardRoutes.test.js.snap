// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes/> Debe mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <span
      className="navbar-brand"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <p
          className="nav-item nav-link m-0"
        >
          Edgar
        </p>
        <button
          className="nav-item nav-link btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </div>
    </div>
  </nav>,
  <div
    className="container-fluid"
  >
    <div
      className="mt-3 container-fluid"
    >
      <div
        className="justify-content-center row"
      >
        <div
          className="col-12"
        >
          <h1>
            Marvel
          </h1>
          <hr />
        </div>
      </div>
      <div
        className="justify-content-center row"
      >
        <div
          className="col-12"
        >
          <div
            className="animate__animated animate__fadeIn card-columns"
          >
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Spider Man"
                    className="card-img-top"
                    src="./assets/heroes/marvel-spider.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Spider Man
                    </div>
                    <p
                      className="card-text"
                    >
                      Peter Parker
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Amazing Fantasy #15
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-spider"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Captain America"
                    className="card-img-top"
                    src="./assets/heroes/marvel-captain.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Captain America
                    </div>
                    <p
                      className="card-text"
                    >
                      Steve Rogers
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Captain America Comics #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-captain"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Iron Man"
                    className="card-img-top"
                    src="./assets/heroes/marvel-iron.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Iron Man
                    </div>
                    <p
                      className="card-text"
                    >
                      Tony Stark
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Tales of Suspense #39
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-iron"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Thor"
                    className="card-img-top"
                    src="./assets/heroes/marvel-thor.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Thor
                    </div>
                    <p
                      className="card-text"
                    >
                      Thor Odinson
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Journey into Myster #83
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-thor"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Hulk"
                    className="card-img-top"
                    src="./assets/heroes/marvel-hulk.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Hulk
                    </div>
                    <p
                      className="card-text"
                    >
                      Bruce Banner
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Incredible Hulk #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hulk"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Wolverine"
                    className="card-img-top"
                    src="./assets/heroes/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Wolverine
                    </div>
                    <p
                      className="card-text"
                    >
                      James Howlett
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Incredible Hulk #180
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-wolverine"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Daredevil"
                    className="card-img-top"
                    src="./assets/heroes/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Daredevil
                    </div>
                    <p
                      className="card-text"
                    >
                      Matthew Michael Murdock
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Daredevil #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-daredevil"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Hawkeye"
                    className="card-img-top"
                    src="./assets/heroes/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Hawkeye
                    </div>
                    <p
                      className="card-text"
                    >
                      Clinton Francis Barton
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        Tales of Suspense #57
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-hawkeye"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Cyclops"
                    className="card-img-top"
                    src="./assets/heroes/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Cyclops
                    </div>
                    <p
                      className="card-text"
                    >
                      Scott Summers
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        X-Men #1
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-cyclops"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-5"
                >
                  <img
                    alt="Silver Surfer"
                    className="card-img-top"
                    src="./assets/heroes/marvel-silver.jpg"
                  />
                </div>
                <div
                  className="col-7"
                >
                  <div
                    className="card-body"
                  >
                    <div
                      className="card-title h5"
                    >
                      Silver Surfer
                    </div>
                    <p
                      className="card-text"
                    >
                      Norrin Radd
                    </p>
                    <p
                      className="card-text"
                    >
                      <small
                        className="text-muted"
                      >
                        The Fantastic Four #48
                      </small>
                    </p>
                    <a
                      href="/hero/marvel-silver"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-outline-primary"
                        disabled={false}
                        type="button"
                      >
                        Mas...
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
